pkgs/development/beam-modules/build-rebar3.nix

```diff
@@ -52,29 +49,19 @@ let
       rm -f rebar rebar3
     '' + postPatch;
 
-    configurePhase = ''
-      runHook preConfigure
-      ${erlang}/bin/escript ${rebar3.bootstrapper} ${debugInfoFlag}
-      runHook postConfigure
-    '';
-
     buildPhase = ''
       runHook preBuild
-      HOME=. rebar3 compile
-      ${if compilePorts then ''
-        HOME=. rebar3 pc compile
```

Comment:
I agree that packages should use the hooks, but IIRC a lot of packages in hex.pm in ~2015 did not do that in practice. Also the `provider_hooks` recommended in the `port_compiler` README won't trigger for `rebar3 bare compile`, since it is not the `rebar3 compile` which everyone sets the hooks for.

Do you know if we can use `rebar3 bare compile` and `rebar3 pc compile` together at all? If so, maybe keeping `compilePorts` flag as an option might be a good middle ground.

IMO the best solution is to introduce a `REBAR3_HERMETIC` flag of sorts into upstream of rebar3 so we can use it. I tried and failed to convince rebar maintainers to make it so.

Second best might be to maintain an always-included rebar3 patch, so that we can set the flag only inside of `buildRebar3`.

Reply:
Fortunately it looks like `rebar3 bare compile` still triggers the `provider_hooks`! I tested a few different nif-containing packages below (some use `pc`, others like `katipo` and `jiffy` have their own custom compilation) and it all seems to work, I'm quite surprised, so I think we can drop `compilePorts`, at this stage I think if we encounter any packages for which it doesn't work, the best approach would be to fix those specific packages

Reply:
This is unexpected and great news!